syntax = "proto3";

package syft.lib.adp;
import "proto/core/common/common_object.proto";
import "proto/lib/adp/entity.proto";

message Scalar {
  syft.core.common.UID id = 1;
  bool has_name = 2;
  string name = 3;
  bool has_value = 4;
  float value = 5;
  bool has_min_val = 6;
  float min_val = 7;
  bool has_max_val = 8;
  float max_val = 9;
  // bool has_poly = 10;
  // Symbol poly = 11;
  bool has_entity = 12;
  Entity entity = 13;
  bool enabled = 14;
}
